### PR TITLE
Use the updated features  on the matomo-tracker package to improve an…

### PR DESCRIPTION
**Pull request by royderks**
…alytics tracking  for the project

### CODE REVIEW

File: src/app/pages/ArticleDetailPage/ArticleDetailPage.jsx [hunk 1]
```diff
@@ -111,6 +111,12 @@ const StyledAccordion = styled(Accordion)`
   margin-top: ${themeSpacing(2)};
 `
 
+const StyledLink = styled(Link)`
+  & > * {
```
Comment (royderks) [inline src/app/pages/ArticleDetailPage/ArticleDetailPage.jsx:115]: Should we update in asc-ui? We want clicks on the link to be registered on a click on the link, and not by a link on the inner elements and have the click event bubble. The link should be the only clickable element, unless specifically overwritten by the user of the library

Comment (royderks) [reply src/app/pages/ArticleDetailPage/ArticleDetailPage.jsx:115]: @chrisvanmook @jonkoops

Comment (jonkoops) [reply src/app/pages/ArticleDetailPage/ArticleDetailPage.jsx:115]: Seems a bit like a hack IMHO, why isn't the analytics handling this properly?

File: src/app/AppBody.jsx [hunk 2]
```diff
@@ -42,6 +43,10 @@ const AppBody = ({
   embedPreviewMode,
 }) => {
   const query = useSelector(getQuery)
+
+  const { enableLinkTracking } = useMatomo()
```
[the remark this answers is not in the document]
Comment (royderks) [reply src/app/AppBody.jsx:47]: The useEffect is already handled by the package, where the event handler is added.

File: src/app/pages/ArticleDetailPage/ArticleDetailPage.jsx (hunk 1 above)
Comment (royderks) [reply src/app/pages/ArticleDetailPage/ArticleDetailPage.jsx:115]: I believe it's due to how the application is being rendered, that the event can't be caught. The target element is the inner span while the currentTarget is the entire DOM (without the rendered html elements rendered by React)

File: src/app/AppBody.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply src/app/AppBody.jsx:47]: Right, but we still do not want to call `enableLinkTracking` on every render right?

[the remark this answers is not in the document]
Comment (royderks) [reply src/app/AppBody.jsx:47]: It doesn't really matter, as the `AppBody` component is rerendered on every route change, it will be called either way

**Pull request merged**